src/Middleware/RequestThrottling/test/SemaphoreWrapperTests.cs

```diff
@@ -10,6 +10,13 @@
 
 namespace Microsoft.AspNetCore.RequestThrottling.Tests
 {
+    public static class TaskExtensions
```

Comment:
Nit: put this in its own file

Reply:
Copy SignalR's TaskExtensions instead like @anurse suggests and we'll look at sharing the source later.